CSHARP-744 Support legacy empty column names

joao-r-reis opened the pull request:
From CSHARP-744 description:

> Usually {{""}} is not a valid column name, but it can occasionally appear when migrating Thrift tables with COMPACT STORAGE to modern storage tables. 

Metadata API (i.e. `GetTable`) already supports this but LINQ and Mapper don't.

jorgebay commented on src/Cassandra.IntegrationTests/Core/EmptyColumnTests.cs:
```diff
@@ -0,0 +1,275 @@
+                    when = new
+                    {
+                        query = string.Format(
+                            "SELECT * FROM system_schema.tables WHERE table_name='{0}' AND keyspace_name='{1}'",
```
Nit: we could use string interpolation that is less error prone.

joao-r-reis replied:
done

jorgebay commented on src/Cassandra.IntegrationTests/Core/EmptyColumnTests.cs:
```diff
@@ -0,0 +1,275 @@
+                    }
+                    finally
+                    {
+                        var a = "";
```
Probably leftover?

joao-r-reis replied:
oops

joao-r-reis replied:
done

jorgebay commented on src/Cassandra.IntegrationTests/Core/EmptyColumnTests.cs:
```diff
@@ -0,0 +1,275 @@
+                        Assert.IsNotNull(tableMetadata);
+
+                        var rs = session.Execute("SELECT \"\", \" \" FROM testks.testtable");
+                        var row = rs.SingleOrDefault();
```
NIT: We could use `Single()` that throws when there are no values, and avoid null conditional operators below.

joao-r-reis replied:
done

jorgebay commented on src/Cassandra.IntegrationTests/Core/EmptyColumnTests.cs:
```diff
@@ -0,0 +1,275 @@
+                        var tableMetadata = session.Cluster.Metadata.GetTable("testks", "testtable");
+                        Assert.IsNotNull(tableMetadata);
+
+                        var rs = session.Execute("SELECT \"\", \" \" FROM testks.testtable");
```
Can we include a prepare statement execution?

joao-r-reis replied:
good idea, added

joao-r-reis commented:
Forgot to prime the queries for C* < 3.0.0, added them in a separate test. Also changed the test filter in `build.yaml` to match the one being used in appveyor.

jorgebay commented on src/Cassandra.IntegrationTests/Core/EmptyColumnTests.cs:
```diff
@@ -0,0 +1,544 @@
+
+namespace Cassandra.IntegrationTests.Core
+{
+    [TestFixture]
```
In any case, I think we should add it to the `short` category.

joao-r-reis replied:
Fair enough, I'll do it
